File: leetcode/top-interview-questions-easy/strings/8. longest_common_prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  var prefix = strs[0];
  if (strs.length === 1) {
    return prefix;
  }

  for (var i = 1; i < strs.length; i++) {
    var current = strs[i];

    var newPrefix = "";
    for (var j = 0; j < current.length; j++) {
      if (current[j] === prefix[j]) {
        newPrefix += current[j];
      } else {
        break;
      }
    }

    if (newPrefix === "") {
      return newPrefix;
    }

    prefix = newPrefix;
  }

  return prefix;
};

File: leetcode/top-interview-questions-easy/strings/8. longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0]
        if (len(strs) == 1):
            return prefix
        
        for i in range(len(strs)):
            current = strs[i]
            newPrefix = ''
            
            limit = min(len(prefix), len(current))
            for j in range(limit):
                if current[j] == prefix[j]:
                    newPrefix += current[j]
                else:
                    break
            
            if newPrefix == "":
                return newPrefix
            
            prefix = newPrefix
        
        return prefix
        
       